create web config and create first controller

maksimk99 opened the pull request.

maksimk99 committed:
create rest module and first controller

maksimk99 commented:
I will add documentation for requests later

maksimk99 commented on dao/src/main/java/com/carhouse/dao/impl/CarDaoImpl.java:
```diff
@@ -0,0 +1,152 @@
+    public Car getCar(final int index) {
+        SqlParameterSource parameters = new MapSqlParameterSource()
+                .addValue("id", index);
+        LOGGER.debug("method getCar with parameter: {} was called", index);
```
`method getCar with parameter: [{}] was called`

maksimk99 commented on rest/src/main/webapp/index.html:
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
```
should be removed because it's  REST application

maksimk99 commented on service/src/main/java/com/carhouse/service/impl/FuelTypeServiceImpl.java:
```diff
@@ -0,0 +1,95 @@
+ * The class provides methods to manage FuelType model.
+ *
+ * @author Katuranau Maksimilyan
+ */
```
`{@see FuelTypeService}`

maksimk99 commented on service/src/main/java/com/carhouse/service/impl/FuelTypeServiceImpl.java:
```diff
@@ -0,0 +1,95 @@
+    /**
+     * Logger.
+     */
+    private static final Logger LOGGER = LogManager.getLogger(FuelTypeServiceImpl.class);
```
replace it before constructor

maksimk99 reviewed (commented):
Add tests: REST and service layer

maksimk99 committed:
add some changes to JavaDoc and project structure

aleksieiev commented on dao/src/test/java/com/carhouse/dao/impl/CarDaoImplTest.java:
```diff
@@ -0,0 +1,101 @@
+    }
+
+    @Test
+    void addCarCharacteristics() {
```
rename method

aleksieiev reviewed (commented):
Add tests please

maksimk99 commented:
I have one question:
What method of testing rest controllers to choose: Mock or WireMock?
I think Mock is enough for it.

aleksieiev commented:
Why do you think so?

maksimk99 commented:
I was wrong, WireMock is better
Wiremock provides a simulator for HTTP-based APIs while Mockito provides the mock implementation of the method/object.
When we are using Mockito, we can't test any code to invoke REST service, getting HTTP response and deserializing response. 
It means WireMock is better then Mockito for testing rest controllers.

aleksieiev commented:
By the way, what are we going to test?

maksimk99 committed:
add documentation for rest endpoints and some tests

maksimk99 committed:
change documentation for rest endpoints

maksimk99 commented:
I've deleted some endpoints.
I will change the database structure for carCharacteristic model and add more changes to endpoints,
but it will be another issue

maksimk99 committed:
change documentation for rest endpoints style

maksimk99 commented:
I've opened an issue to choose a way of testing REST controllers
see `#12`

aleksieiev commented:
https://github.com/java-fat-unicorn-team/CarSale/issues/12
MockMVC has been chosen

aleksieiev reviewed (changes requested):
No changes has been found.

maksimk99 committed:
add test for rest controller

aleksieiev reviewed (commented):
Great :-)
Does this PR require anything additional? I remember something about service tests.

maksimk99 commented:
I've added these tests before.
I have one question: if I've removed add, update and delete endpoints for FuelType model, should I remove these operations from DAO layer?

aleksieiev commented:
How do you think?

maksimk99 commented:
remove

aleksieiev commented:
So. be brave and remove them if it isn't required.

maksimk99 committed:
remove redundant methods for FuelType model on DAO and service layers

aleksieiev commented on service/src/main/test/java/com/carhouse/service/impl/FuelTypeServiceImplTest.java:
```diff
@@ -0,0 +1,73 @@
+    private List<FuelType> listFuelTypes;
+    @BeforeEach
+    void addFuelTypesToService() {
+        listFuelTypes = new ArrayList<>(){{
```
can be replaced to 
` private List<FuelType> listFuelTypes = new ArrayList<>(){{ ... }};`

aleksieiev commented on service/src/main/test/java/com/carhouse/service/impl/FuelTypeServiceImplTest.java:
```diff
@@ -0,0 +1,45 @@
+
+    @BeforeEach
+    void addFuelTypesToService() {
+        listFuelTypes = new ArrayList<>() {{
```
can be moved to `private` initialization
